File: train/lib/solarModel/matlab/zentihAngle.m
function [ z,cosz ] = zentihAngle( lat,long,DOY,hora )
%UNTITLED5 Summary of this function goes here
%   Detailed explanation goes here

 d = declination(DOY);
 
 % assuming spain -> LSTM = 0
 hra =  hourAngle( DOY, hora, long, 0 );
 
 %cosz = cosd(lat)*cosd(d)*cosd(hra)+sind(lat)*sind(d);
 cosz = cosd(lat)*cosd(d)*cosd(hra) + sind(lat)*sind(d);
 z = acosd(cosz);
 
 %%Correccion nacho: si el angulo es mayor de 90, lo dejamos en 90
 %% porque querra decir que es de noche
 if (abs(z) > 90) 
     z=90;
     cosz=0;
 end
end

File: train/lib/info_riego_data/lib/solar_model/zentihAngle.py
import trigonometry as tr
import math
from declination import declination
from hourAngle import hourAngle

def zentihAngle(lat, lng, doy, hora):

	d = declination(doy) #in radians

        hra = hourAngle(doy, hora, lng, 0) #in radians

	cosz = tr.cosd(lat) * tr.cosd(d) * tr.cosd(hra) + tr.sind(lat) * tr.sind(d)

	z = tr.acosd(cosz)

	if(abs(z) > 90):
		z = 90
		cosz = 0

	return z, cosz
